centinel/primitives/dnslib.py

```diff
@@ -15,7 +15,7 @@ def get_ips(host, nameserver=None, record="A"):
     return lookup_domain(host, nameservers=nameservers, rtype=record)
 
 
-def lookup_domain(domain, nameservers=[], rtype="A", timeout=10):
+def lookup_domain(domain, nameservers=[], rtype="A", timeout=2):
```

Comment:
I think that 2 seconds should be sufficient, but is not guaranteed to be. Some regions (think Africa) typically have RTTs in the range of 400-500ms. If this is a query over TCP (common for EDNS options), then you have two RTTs for the query, which could easily be over 1 second. It would be odd, but I could totally see congestion issues leading to longer timeouts. In our case, the timeout is defined for the socket, but I think it is possible for the timeout to exceed 2 seconds.

Ultimately, the average or typical case really doesn't matter in this case. If we are going to make that claim that a DNS request is being dropped based on this timeout, we cannot settle for an inconclusive measurement-> 2 second is not conclusive enough for me. In these odd cases, we *need* to wait longer to differentiate packet drops and longer delays. If performance is a concern, then I also built a parallel version of this, so it really doesn't matter.

tl;dr We are making claims about censorship based on these queries, so we cannot afford uncertainty. We must have a longer timeout